ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -340,6 +341,23 @@ ExecutorService getScheduler() {
   @Override
   protected void updateResolvedContainers(Account account, Collection<Container> resolvedContainers)
       throws AccountServiceException {
+
+    // Check for name/id/version conflicts for containers being updated with those in local cache.
+    infoMapLock.readLock().lock();
+    try {
+      if (accountInfoMapRef.get()
+          .hasConflictingContainer(resolvedContainers, account.getId(), config.ignoreVersionMismatch)) {
+        logger.error(
+            "Containers={} under Account={} conflict with the containers in local cache. Cancel the update operation.",
+            account.getAllContainers(), account.getId());
```

Comment:
minor: can you print out  account name instead of account id?  
Also `account.getAllContainers()` can be replaced with `resolvedContainers`